graph-builder/src/config/cli.rs

```diff
@@ -0,0 +1,83 @@
+//! Command-line options.
+
+// TODO(lucab): drop all aliases after migration.
+
+/// CLI configuration flags, top-level.
+#[derive(Debug, StructOpt)]
+pub struct CliOptions {
+    /// Verbosity level
+    #[structopt(long = "verbosity", short = "v", parse(from_occurrences))]
+    pub verbosity: u8,
+
+    /// Path to configuration file
+    #[structopt(short = "c")]
+    pub config_path: Option<String>,
+
+    #[structopt(flatten)]
+    pub service: ServiceOptions,
+
+    #[structopt(flatten)]
+    pub status: StatusOptions,
+
+    #[structopt(flatten)]
+    pub registry: UpstreamRegistryOptions,
+}
+
+/// CLI configuration flags, status service.
+#[derive(Debug, StructOpt)]
+pub struct StatusOptions {
+    /// Address on which the status service will listen
+    #[structopt(long = "status.address")]
+    pub address: Option<String>,
+
+    /// Port to which the status service will bind
+    #[structopt(long = "status.port")]
+    pub port: Option<u16>,
+}
+
+/// CLI configuration flags, main Cincinnati service.
+#[derive(Debug, StructOpt)]
+pub struct ServiceOptions {
+    /// Address on which the server will listen
+    #[structopt(long = "service.address", alias = "address")]
+    pub address: Option<String>,
+
+    /// Port to which the server will bind
+    #[structopt(long = "service.port", alias = "port")]
+    pub port: Option<u16>,
+
+    /// Namespace prefix for all service endpoints
+    #[structopt(long = "service.path_prefix")]
+    pub path_prefix: Option<String>,
+
+    /// Comma-separated set of mandatory client parameters
+    #[structopt(long = "service.mandatory_client_parameters")]
+    pub mandatory_client_parameters: Option<String>,
+}
+
+/// CLI configuration flags, Docker-registry fetcher.
+#[derive(Debug, StructOpt)]
+pub struct UpstreamRegistryOptions {
```

Comment:
I don't plan to reuse this; however, I'm bad at naming things :(

This is technically a "DockerRegistryV2" upstream, and there are likely a "[OciDistributionVx](https://github.com/opencontainers/distribution-spec)" and a "FedoraCoreos" coming up at some point in the future. I went for `registry` only because it was shorter for options.